packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPickerUtils/useSvgParser.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'react';
+
+import { traverseSvg } from './SvgParserHelper';
+
+interface IConvertComponentProps {
+  selectedColor: string;
+  iconString: string;
+}
+
+export const useSvgParser = ({ selectedColor, iconString }: IConvertComponentProps) => {
+  const [svgCode, setSvgCode] = useState<string>('');
+
+  useEffect(() => {
+    try {
+      const decodeBase64 = (str: string) => {
+        if (typeof atob === 'function') {
+          return atob(str);
+        } else {
+          return Buffer.from(str, 'base64').toString('binary');
```

Comment:
This checks useful to ensure that atob function is available and can be called before actually using it. If the check returns true, it means that atob function is defined and can be safely used to decode the string.

Reply:
When may it not be available? Is it not available in all browsers or is it a node/browser difference? Or something else?

Reply:
The `atob` is typically used in javascript to decode strings. While it is a standard function in all modern browsers, it may not available in all older browsers. eg- IE before IE(10), opera mini etc.